x-pack/legacy/plugins/alerting/server/lib/alerts_client_factory.test.ts

```diff
@@ -93,6 +93,16 @@ test('createAPIKey() returns { created: false } when security is disabled', asyn
   expect(createAPIKeyResult).toEqual({ created: false });
 });
 
+test('createAPIKey() returns { created: false } when security is enabled but ES security is disabled', async () => {
```

Comment:
I think it would be a good idea to add a functional test around this to test what happens when `createAPIKey` throws an error, as the context seems to be one where we need to ensure we gracefully handle it.

Reply:
Fixed in https://github.com/elastic/kibana/pull/51639/commits/4d30e0ff1ca3b4b8bf9d461f972809a8ed7e8d9e. It is worth noting we won't gracefully handle errors thrown. The function will make sure errors are passed along. This is in the scenario TLS is disabled and we can't allow alerts being created without API keys (bad environment setup and will be handled here: https://github.com/elastic/kibana/issues/49403).